aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -216,7 +216,46 @@ Tensor& cumprod_out(Tensor& result, const Tensor& self, int64_t dim, c10::option
   return result;
 }
 
-
+std::tuple<Tensor&, Tensor&> cummax_out(Tensor& out, Tensor& indices, const Tensor& self, int64_t dim, c10::optional<ScalarType> dtype) {
+  TORCH_CHECK(
+      !dtype.has_value() || (out.scalar_type() == dtype.value()),
+      "provided dtype must match dtype of result in cummax. Got ",
+      toString(out.scalar_type()),
+      " and ",
+      toString(dtype.value()),
+      ".");
+   {
+    NoNamesGuard guard;
+    if(self.dim() == 0) {
+      out = self;
```

Comment:
you almost certainly don't want to do this (we have a longstanding action item to change the output type from `Tensor&` so folks don't make this mistake).

The semantics of "out" are that you treat the passed-in tensor as a buffer, but here you are just reassigning the reference, i..e. you are totally ignoring the buffer.  Also indices is set to self, but that's not the correct dtype is it?

Reply:
thanks for pointing it out! updated